ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/sharing/AppSharingPresenter.java

```diff
@@ -0,0 +1,341 @@
+import org.iplantc.de.client.util.JsonUtil;
+import org.iplantc.de.collaborators.client.util.CollaboratorsUtil;
+import org.iplantc.de.commons.client.ErrorHandler;
+import org.iplantc.de.commons.client.info.IplantAnnouncer;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.json.client.JSONString;
+import com.google.gwt.json.client.JSONValue;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.HasOneWidget;
+
+import com.sencha.gxt.core.shared.FastMap;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AppSharingPresenter implements SharingPresenter {
+
+    private final class LoadPermissionsCallback implements AsyncCallback<String> {
+        private final class GetUserInfoCallback implements AsyncCallback<FastMap<Collaborator>> {
+            private final List<String> usernames;
+
+            private GetUserInfoCallback(List<String> usernames) {
+                this.usernames = usernames;
+            }
+
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(FastMap<Collaborator> results) {
+                dataSharingMap = new FastMap<>();
+                for (String userName : usernames) {
+                    Collaborator user = results.get(userName);
+                    if (user == null) {
+                        user = collaboratorsUtil.getDummyCollaborator(userName);
+                    }
+
+                    List<Sharing> dataShares = new ArrayList<>();
+
+                    dataSharingMap.put(userName, dataShares);
+
+                    for (JSONObject share : sharingList.get(userName)) {
+                        String path = jsonUtil.getString(share, "path"); //$NON-NLS-1$
+                        Sharing dataSharing = new Sharing(user,
+                                                          buildPermissionFromJson(share),
+                                                          path,
+                                                          DiskResourceUtil.getInstance()
+                                                                          .parseNameFromPath(path));
+                        dataShares.add(dataSharing);
+                    }
+                }
+
+                permissionsPanel.loadSharingData(dataSharingMap);
+                permissionsPanel.unmask();
+            }
+        }
+
+        @Override
+        public void onFailure(Throwable caught) {
+            permissionsPanel.unmask();
+            ErrorHandler.post(caught);
+        }
+
+        @Override
+        public void onSuccess(String result) {
+            JSONArray permissionsArray =
+                    jsonUtil.getArray(jsonUtil.getObject(result), "apps"); //$NON-NLS-1$
+            if (permissionsArray != null) {
```

Comment:
I made some comments in this direction.

On Mon, Feb 1, 2016, 9:33 PM Ash <notifications@github.com> wrote:

> In
> ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/sharing/AppSharingPresenter.java
> <https://github.com/cyverse/DE/pull/60#discussion_r51522705>:
>
> > +                permissionsPanel.loadSharingData(dataSharingMap);
> > +                permissionsPanel.unmask();
> > +            }
> > +        }
> > +
> > +        @Override
> > +        public void onFailure(Throwable caught) {
> > +            permissionsPanel.unmask();
> > +            ErrorHandler.post(caught);
> > +        }
> > +
> > +        @Override
> > +        public void onSuccess(String result) {
> > +            JSONArray permissionsArray =
> > +                    jsonUtil.getArray(jsonUtil.getObject(result), "apps"); //$NON-NLS-1$
> > +            if (permissionsArray != null) {
>
> I'm probably missing something, but instead of String couldn't this be a
> type of bean like App or AppList? It would eliminate the need for all the
> JSONArrays and JSONObject manipulations in this file.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyverse/DE/pull/60/files#r51522705>.
>